bin/network-monitor/src/status.rs

```diff
@@ -304,6 +347,58 @@ async fn check_rpc_status(
 // REMOTE PROVER STATUS CHECKER
 // ================================================================================================
 
+/// Runs a task that continuously checks remote prover status and updates a watch channel.
+///
+/// This function spawns a task that periodically checks a remote prover service status
+/// and sends updates through a watch channel.
+///
+/// # Arguments
+///
+/// * `prover_url` - The URL of the remote prover service.
+/// * `name` - The name of the remote prover.
+/// * `status_sender` - The sender for the watch channel.
+///
+/// # Returns
+///
+/// A `Result` indicating success or failure.
+#[instrument(target = COMPONENT, name = "remote-prover-status-task", skip_all)]
+async fn run_remote_prover_status_task(
```

Comment:
If there is any way to deduplicate the logic in this and `run_rpc_status_task` that would be good. Might not be worth the generics involved, if it is practical at all.

Reply:
I think that we can get away with this by passing the check function as a parameter